src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -166,5 +173,15 @@ export class PythonProjectCreator extends ScriptProjectCreatorBase {
                 await cpUtils.executeCommand(ext.outputChannel, this.functionAppPath, `source ${await this.getVenvActivatePath(Platform.MacOS)} && ${funcInitPython}`);
                 break;
         }
+        // .gitignore is created by `func init`
+        const gitignorePath: string = path.join(this.functionAppPath, gitignoreFileName);
+        if (await fse.pathExists(gitignorePath)) {
+            ext.outputChannel.appendLine(localize('gitAddFunc_Env', 'Adding "{0}" to .gitignore...', this.funcEnv));
+            let gitignoreContents: string = (await fse.readFile(gitignorePath)).toString();
+            // the func_env and ._python_packages are recreated and should not be checked in
+            gitignoreContents = gitignoreContents.concat(`${os.EOL}${this.funcEnv}`);
```

Comment:
Eventually the func cli might do this for us. So you should do some checking like `if (!gitignoreContents.includes(this.funcEnv))`. No need to edit the file if it already has those lines.

Reply:
I did a bit more checking.  Probably is fine, but will give you a chance to verify if you don't like it